### PR TITLE
Compatible with pydantic exporting model's method

### PR DESCRIPTION
**Pull request by xl9211**
Compatible with pydantic exporting model's method

### CODE REVIEW

**Comment by Kludex** [comment]
@xl9211 Can you close this one? It's exactly the same as #1720

File: fastapi/encoders.py [hunk 1]
```diff
@@ -45,10 +45,6 @@ def jsonable_encoder(
             "exclude_unset to keep in line with Pydantic v1, support for it will be "
             "removed soon."
         )
-    if include is not None and not isinstance(include, set):
-        include = set(include)
-    if exclude is not None and not isinstance(exclude, set):
-        exclude = set(exclude)
```
Comment (MashhadiNima) [inline fastapi/encoders.py:51]: Why did you remove these line?
Considering the type of variables, i think there is no possibility of removing these lines.

File: fastapi/encoders.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def generate_encoders_by_class_tuples(
 def jsonable_encoder(
     obj: Any,
     include: Union[SetIntStr, DictIntStrAny] = None,
-    exclude: Union[SetIntStr, DictIntStrAny] = set(),
```
Comment (MashhadiNima) [inline fastapi/encoders.py:33]: When you set "None" as default,  you will need to check condition of existence in the rest of the code.

File: fastapi/encoders.py (hunk 1 above)
Comment (xl9211) [reply fastapi/encoders.py:51]: When exclude is dict, that thing were wrong.
I user pydantic's advanced case.

Also see:
https://pydantic-docs.helpmanual.io/usage/exporting_models/#advanced-include-and-exclude

**Comment by tiangolo** [comment]
Thanks for your interest, it seems you are assuming Pydantic 1.x. I can't remove the backward-compatibility code until I deprecate Pydantic 0.32.0.

I should do that soon, but this can't be done before that.

So I'll pass on this for now. Thanks for your effort! :cake: :coffee:
